pageserver/src/bin/pageserver.rs

```diff
@@ -325,6 +326,10 @@ fn start_pageserver(conf: &'static PageServerConf) -> anyhow::Result<()> {
         );
 
         if let Some(metric_collection_endpoint) = &conf.metric_collection_endpoint {
+            let metrics_ctx = RequestContext::todo_child(
+                TaskKind::MetricsCollection,
+                DownloadBehavior::Error, // metrics collector shouldn't be downloading anything
```

Comment:
Calculating the synthetic size requires information about relation sizes, which can require downloading stuff. But I believe this is correct, because we use this context only for the long-running collector, and create child contexts with DownloadBehavior::Download to do the synthetic size calculations. Let's explain that in the comment here, this is misleading as it is.